blocks/library/tweet/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { registerBlock, query } from '../../api';
+import Sandbox from '../../../components/sandbox';
+import Button from '../../../components/button';
+import Placeholder from '../../../components/placeholder';
+
+const { prop } = query;
+
+registerBlock( 'core/tweet', {
+	title: wp.i18n.__( 'Tweet' ),
+	icon: 'twitter',
+
+	category: 'social',
+
+	attributes: {
+		url: prop( '*', 'innerHTML' ), // our html is just a div with the url in, for WP's oembed to process
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.fetchTweet = this.fetchTweet.bind( this );
+			this.state = {
+				url: this.props.attributes.url,
+				html: '',
+				error: false,
+				fetching: false,
+			};
+		}
+		doFetch( url, setAttributes, setState ) {
+			setState( { fetching: true, error: false } );
+			jQuery.ajax( {
+				type: 'GET',
+				dataType: 'jsonp',
+				data: {},
+				timeout: 5000,
+				url: 'https://publish.twitter.com/oembed?url=' + encodeURI( url ),
+				error: function() {
+					setState( { fetching: false, error: true } );
+				},
+				success: function( msg ) {
+					setAttributes( { url: url } );
```

Comment:
Optional, but one nice feature included with ES2015 is [object shorthand properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015), so this could be written as:

```
this.props.setAttributes( { url } );
```